clouddriver-elasticsearch-aws/src/main/kotlin/com/netflix/spinnaker/clouddriver/elasticsearch/aws/ClusterScalingPolicyTaggingAgent.kt

```diff
@@ -0,0 +1,327 @@
+        .also { clusterPolicies ->
+          for (policy in byServerGroupName.getOrDefault(serverGroup.autoScalingGroupName, emptyList())) {
+            if (clusterPolicies.none { it.isEqual(policy) }) {
+              clusterPolicies.add(policy)
+            }
+          }
+        }
+    }
+
+    return results.filter { it.value.isNotEmpty() }
+  }
+
+  override fun getPollIntervalMillis(): Long = TimeUnit.SECONDS.toMillis(10)
+  override fun getTimeoutMillis(): Long = TimeUnit.SECONDS.toMillis(30)
+  override fun getAgentType(): String = ClusterScalingPolicyTaggingAgent::class.java.simpleName
+  override fun getProviderName(): String = AwsProvider.PROVIDER_NAME
+
+  internal fun List<ScalingPolicy>.toEntityTagValue(): Map<String, Any> =
+    mapOf("value" to objectMapper.convertValue(map { it.toGenericMap(objectMapper) }, LIST_OF_MAPS))
+
+
+  private fun retry(fn: () -> Unit) {
+    retrySupport.retry(fn, 3, 500, false)
+  }
+}
+
+internal typealias IndexedScalingPolicies = Map<String, List<ScalingPolicy>>
+
+internal data class AccountRegionIndex(
+  val credentials: NetflixAmazonCredentials,
+  val region: AWSRegion,
+  val serverGroups: List<AutoScalingGroup>,
+  val scalingPolicies: AccountRegionScalingPolicies,
+  val clusterEntityTags: Map<String, List<EntityTags>>
+)
+
+internal data class AccountRegionScalingPolicies(
+  val all: List<ScalingPolicy>,
+  val byServerGroupName: IndexedScalingPolicies,
+  val byCluster: IndexedScalingPolicies
+)
+
+internal fun AutoScalingGroup.toLog(credentials: NetflixAmazonCredentials, region: AWSRegion) =
+  "${credentials.name}/${region.name}/$autoScalingGroupName"
+
+/**
+ * This isn't exactly scientific, but it's better than just comparing num of scaling policies.
+ */
+internal fun ScalingPolicy.isEqual(scalingPolicy: ScalingPolicy): Boolean {
+  return policyType == scalingPolicy.policyType &&
+    adjustmentType == scalingPolicy.adjustmentType &&
+    minAdjustmentStep == scalingPolicy.minAdjustmentStep &&
+    minAdjustmentMagnitude == scalingPolicy.minAdjustmentMagnitude &&
+    scalingAdjustment == scalingPolicy.scalingAdjustment &&
+    cooldown == scalingPolicy.cooldown &&
+    stepAdjustments.all { a -> scalingPolicy.stepAdjustments.any { b -> a == b } } &&
+    metricAggregationType == scalingPolicy.metricAggregationType &&
+    alarms.size == scalingPolicy.alarms.size &&
+    targetTrackingConfiguration == scalingPolicy.targetTrackingConfiguration
+}
+
+internal fun ScalingPolicy.toGenericMap(objectMapper: ObjectMapper): Map<String, Any?> =
+  objectMapper.convertValue<MutableMap<String, Any?>>(
+    this, object : TypeReference<MutableMap<String, Any?>>() {}
+  ).apply {
+    remove("autoScalingGroupName")
+    remove("policyName")
+    remove("policyARN")
+    set("alarms", alarms.map { mapOf<String, Any?>() })
+  }
+
+/**
+ * This is kinda sad, but Entity Tags are designed to be a map, so if we're storing multiple values, the root-level
```

Comment:
This is likely correctable but just hasn't been all that pressing, potentially as simple as adding a `list` to the `EntityTagValueType` enum (and the associated serialization / deserialization).

```
static enum EntityTagValueType {
    literal, // number or string
    object   // map
  }
```

Reply:
It's not that big of a deal to me, just wanted to add a comment so that future me doesn't forget why I'm doing it (and if it changes in the future, it'll be an easy call out to refactor).